src/payjoin/app.rs

```diff
@@ -0,0 +1,214 @@
+use std::collections::HashMap;
+
+use anyhow::{anyhow, Context, Result};
+use payjoin::{
+    receive::{PayjoinProposal, ProvisionalProposal, UncheckedProposal},
+    Error,
+};
+use tracing::instrument;
+
+type ProtoClient =
+    crate::api::proto::bria_service_client::BriaServiceClient<tonic::transport::Channel>;
+use crate::{
+    app::{error::ApplicationError, App},
+    primitives::bitcoin,
+    profile::Profile,
+};
+
+#[derive(Clone)]
+pub struct PayjoinApp {
+    app: App,
+    profile: Profile,
+}
+
+impl PayjoinApp {
+    pub async fn run(app: App, profile: Profile) -> Self {
+        Self { app, profile }
+    }
+
+    #[instrument(name = "payjoin_app.process_proposal", skip(self), err)]
+    pub async fn process_proposal(
+        self,
+        proposal: UncheckedProposal,
+    ) -> Result<PayjoinProposal, Error> {
+        // in a payment processor where the sender could go offline, this is where you schedule to broadcast the original_tx
+        let _to_broadcast_in_failure_case = proposal.extract_tx_to_schedule_broadcast();
+
+        // The network is used for checks later
+        let network = self.app.network();
+
+        // Receive Check 1: Can Broadcast
+        let proposal = proposal.check_broadcast_suitability(None, |tx| {
+            let _raw_tx = bitcoin::consensus::encode::serialize_hex(&tx);
+            // TODO test_mempool_accept e.g.:
+            //
+            // let mempool_results =
+            //    bitcoind.test_mempool_accept(&[raw_tx]).map_err(|e| Error::Server(e.into()))?;
+            // match mempool_results.first() {
+            //     Some(result) => Ok(result.allowed),
+            //     None => Err(Error::Server(
+            //         anyhow!("No mempool results returned on broadcast check").into(),
+            //     )),
+            // }
+            Ok(true)
+        })?;
+        tracing::trace!("check1");
```

Comment:
What's the most straightforward way to access a `testmempoolaccept` function like bitcoind's?

Reply:
Since bria uses Fulcrum and [Fulcrum does not have such a function](https://github.com/cculianu/Fulcrum/issues/44), we'll need to devise a new way to check mempool acceptance so that we know the original_psbt proposal is acceptable fallback collateral if the sender disappears to prevent probing attacks.

Reply:
I’m okay adding a dependency to bitcoind for this as we want to eventually get rid of fulcrum anyhow.